src/httpClient.ts

```diff
@@ -1,3 +1,5 @@
+import type { BodyInit, Response } from "undici";
+import { fetch, Agent } from "undici";
```

Comment:
Is this using HTTP/2 when possible (so we can do it all on one connection) or HTTP/1.1 (we'd need multiple connections to not get any roundtrip penalty)?

Reply:
we could turn on http2 by passing `enableH2`, it's disabled by default since support is experimental

Reply:
I'm sure it's good enough for us for now, but maybe we should just tinker with that later? If you have that many QPS, the connection pool will probably be really good either way.

Reply:
The undici h2 client will actually log a warning saying it’s experimental, so probably not great to turn on in a prod environment.